mono/metadata/appdomain.c

```diff
@@ -2751,6 +2750,8 @@ typedef struct unload_data {
 static void
 unload_data_unref (unload_data *data)
 {
+	if (!data)
```

Comment:
assert here.  fix the caller (`mono_domain_try_unload` has a new `goto exit` with `thread_data == NULL`).  As far as I can tell none of the other callers ever pass NULL - no need to relax that.

Reply:
`unref(null)` is like `free(null)`. If you disagree, I can add the check at the caller.